bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/widgets/DirectoryDialog.java

```diff
@@ -182,7 +202,13 @@ public String open() {
 		fileDialog.Release();
 	}
 
-	return directoryPath;
+	if (hr == COM.S_OK) {
+		return FileDialogResult.createForSuccessfulSelection(directoryPath);
+	} else if (hr == OS.HRESULT_FROM_WIN32(OS.ERROR_CANCELED)) {
+		return FileDialogResult.createForUserCancellation();
+	} else {
+		return FileDialogResult.createForUnkownError();
+	}
```

Comment:
fwiw: this is not a switch because the cancel state HRESULT is not a constant but a calculated value.